packages/app/src/components/time-series-chart/time-series-chart.tsx

```diff
@@ -393,6 +393,15 @@ export function TimeSeriesChart<
                 config={x}
               />
             ))}
+            {timeAnnotations?.map((x, index) => (
+              <TimeAnnotation
+                key={index}
+                domain={xScale.domain() as [number, number]}
```

Comment:
I don't think we should cast this to a tuple of numbers because the array could hold more than 2 numbers. It's typed like `number[]` on purpose.
```suggestion
                domain=[first({xScale.domain()), last({xScale.domain())]}
```

Reply:
Please expand your code suggestion to also cover the `TimespanAnnotation` component, this also takes the domain as a number tuple. That's where I copied this code from. ;)

Reply:
Ay yes might be better to turn it around;
```tsx
export function TimespanAnnotation({
  start,
  end,
  domain,
  getX,
  height,
}: {
  start: number;
  end: number;
  domain: number[];
  height: number;
  getX: GetX;
}) {
  const min = first(domain);
  const max = last(domain);
  assert(
    typeof min === 'number' && typeof max === 'number',
    `received invalid domain ${domain}`
  );

Reply:
Addendum, while it is, in theory, possible there's a domain consisting of more than two values. In the time-series-chart all the domains are explicitly created with two values. So, I'm not sure if this changes is really that necessary...

Reply:
I think the domain check is a good one though, that definitely simplifies the code, so I'm using that

Reply:
Oops, no, I thought your code suggestion was valid. But now I realize that the values are already being clipped. As the comment states, this allows for Infinity to be passed in as an end date.